File: src/main/java/at/searles/parsing/printing/StringOutStream.java
package at.searles.parsing.printing;

public class StringOutStream implements OutStream {

    private StringBuilder sb = new StringBuilder();

    @Override
    public void append(CharSequence seq) {
        sb.append(seq);
    }

    @Override
    public void append(int codePoint) {
        sb.appendCodePoint(codePoint);
    }

    @Override
    public String toString() {
        return sb.toString();
    }
}

File: src/main/kotlin/at/searles/parsing/printer/StringOutStream.kt
package at.searles.parsing.printer

class StringOutStream : OutStream {
    private val sb = StringBuilder()

    override fun append(seq: CharSequence) {
        sb.append(seq)
    }

    override fun append(codePoint: Int) {
        sb.appendCodePoint(codePoint)
    }

    override fun toString(): String {
        return sb.toString()
    }
}